### ISSUE TITLE
Removing empty imports of modules with moduleSideEffects: false

### ISSUE THREAD

**Issue by stropho** [rollup/rollup#3057]
- Rollup Version: 1.19.4
- Operating System (or Browser): Linux, Mac
- Node Version: 10

### How Do We Reproduce?
I'm not sure if this is a bug or feature request. See the example [REPL](https://rollupjs.org/repl/?version=1.19.4&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiolMjBFWFRFUk5BTCUyMElNUE9SVFMlNUNuJTIwJTIwJTIwWW91JTIwY2FuJTIwaW1wb3J0JTIwZXh0ZXJuYWwlMjBtb2R1bGVzJTIwaW50byUyMHlvdXIlMjBidW5kbGUlMjAlRTIlODAlOTMlNUNuJTIwJTIwJTIwaXQlMjBkb2Vzbid0JTIwbWF0dGVyJTIwaWYlMjB0aGV5J3JlJTIwRVMyMDE1JTIwb3IlMjBsZWdhY3klMjAqJTJGJTVDbmltcG9ydCUyMCU3QkNvbXBvbmVudCU3RCUyMGZyb20lMjAnLiUyRmNvbXBvbmVudCclM0IlNUNuJTVDbmNvbnNvbGUubG9nKENvbXBvbmVudCklMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtYWluMi5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJpbXBvcnQlMjAlN0JDb21wb25lbnQlN0QlMjBmcm9tJTIwJy4lMkZjb21wb25lbnQnJTNCJTVDbiU1Q25jb25zb2xlLmxvZyhDb21wb25lbnQpJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyY29tcG9uZW50LmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QnRvb2wlN0QlMjBmcm9tJTIwJ3NvbWUtcHJpdmF0ZS1tb2R1bGUnJTNCJTVDbiU1Q25leHBvcnQlMjBjb25zdCUyMENvbXBvbmVudCUyMCUzRCUyMCh4KSUyMCUzRCUzRSUyMHRvb2woeCklMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXNtJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

There is an import of external module 
`import { tool } from 'some-private-module';` 
in  code (a shared chunk) shared across multiple entry points. 
Now the entry points contain
`import 'some-private-module';`

In our codebase, we created a rollup plugin with a `resolveId` hook which resolves our ` 'some-private-module'` as 
```
  {
      id: '...some id...',
      external: true,
      moduleSideEffects: false, // tried but didn't help
}
```
Correct me if I'm wrong, but I don't see a good reason why `import 'some-private-module';` exists in the entry points. 

### Expected Behavior
no `import 'some-private-module';` in entry points

### Actual Behavior
`import 'some-private-module';` in entry points exist

**Comment by lukastaegert** [comment]
The `moduleSideEffects: false` logic only prevents external imports from being included if no module in the graph actually uses something that is imported. As soon as there is a single module in the graph that uses the external import, all other imports from this module are included as well, albeit maybe as "empty" imports, which is what you witnessed.

Suppressing those imports would indeed be a non-trivial feature request as the current implementation just works via changing the side-effect detection algorithm, but "being used" is always considered a side-effect.

**Comment by lukastaegert** [comment]
It could be argued, though, that for external imports the problem is a little simpler as we could run an additional check prior to finalization that filters out side-effect-free empty external imports.

**Comment by stropho** [comment]
Wow that was a swift response.
Well we figured out some workaround.
Anyway, let me know @lukastaegert if that additional check is something that could be considered and implemented.
Closing the issue for now

**Comment by lukastaegert** [comment]
This will be fixed by #3369

**Comment by lukastaegert** [comment]
Solved in rollup@2.0.0
